cmd/machine-api-operator/start.go

```diff
@@ -73,10 +73,11 @@ func runStartCmd(cmd *cobra.Command, args []string) {
 	stopCh := make(chan struct{})
 
 	leaderelection.RunOrDie(context.TODO(), leaderelection.LeaderElectionConfig{
-		Lock:          CreateResourceLock(cb, componentNamespace, componentName),
-		LeaseDuration: util.LeaseDuration,
-		RenewDeadline: util.RenewDeadline,
-		RetryPeriod:   util.RetryPeriod,
+		Lock:            CreateResourceLock(cb, componentNamespace, componentName),
+		ReleaseOnCancel: true,
```

Comment:
I don't expect this to do anything, because the `context.TODO()` getting passed in above will never be canceled.  For comparison, the cluster-version operator [sets up a few separate contexts and a TERM catcher][1] so we can:

1. Catch the TERM and cancel `runContext`.
2. Collect all the goroutines that had been using `runContext`, to ensure we no longer have any goroutines managing in-cluster operands.
3. Cancel `postMainContext`, which was used for informers and the leader lease.
4. `shutdownContext`, which lets callers know when they can give up on a graceful child-goroutine reap and stop blocking on it.

[1]: https://github.com/openshift/cluster-version-operator/blob/0ec39d9b2ab1feee8815d7b6b4bbe2db23daf847/pkg/start/start.go#L176-L195